File: spring-cloud-openfeign/feign-provider/feign-provider-biz/src/main/java/com/livk/provider/biz/mapper/UserMapper.java
package com.livk.provider.biz.mapper;

import com.livk.provider.api.domain.Users;
import org.apache.ibatis.annotations.*;

import java.util.List;

/**
 * <p>
 * UserRepository
 * </p>
 *
 * @author livk
 * @date 2021/12/6
 */
@Mapper
public interface UserMapper {

    @Select("select * from users")
    List<Users> selectList();

    @Insert("insert info users (username,password) values (#{username},#{password})")
    int insert(Users users);

    @Delete("delete from users where id = #{id}")
    int deleteById(@Param("id") Long id);
}

File: spring-cloud-openfeign/feign-provider/feign-provider-biz/src/main/kotlin/com/livk/provider/biz/mapper/UserMapper.kt
package com.livk.provider.biz.mapper

import com.livk.provider.api.domain.Users
import org.apache.ibatis.annotations.Delete
import org.apache.ibatis.annotations.Insert
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import org.apache.ibatis.annotations.Select

/**
 * @author livk
 */
@Mapper
interface UserMapper {
    @Select("select * from users")
    fun selectList(): List<Users>

    @Insert("insert info users (username,password) values (#{username},#{password})")
    fun insert(users: Users): Int

    @Delete("delete from users where id = #{id}")
    fun deleteById(@Param("id") id: Long): Int
}
